Marlin/src/pins/esp32/pins_MM_JOKER.h

```diff
@@ -0,0 +1,258 @@
+
+//
+// Heaters / Fans
+//
+#define HEATER_0_PIN                         141  
+#define FAN_PIN                              143  
+#define HEATER_BED_PIN                       140  
+
+//#define CONTROLLER_FAN_PIN                   -1
+#define E0_AUTO_FAN_PIN                    142  // need to update Configuration_adv.h @section extruder
+
+//
+// MicroSD card
+//
+#define SD_MOSI_PIN                           23
+#define SD_MISO_PIN                           19
+#define SD_SCK_PIN                            18
+#define SDSS                                   5
+#define USES_SHARED_SPI                           // SPI is shared by SD card with TMC SPI drivers
+
+//////////////////////////
+// LCDs and Controllers //
+//////////////////////////
+
+#if HAS_WIRED_LCD
+
+  //#define LCD_PINS_RS                         13
+  //#define LCD_PINS_ENABLE                     17
+  //#define LCD_PINS_D4                         16
+
+  #if ENABLED(CR10_STOCKDISPLAY)
+
+    #define BEEPER_PIN                       151
+
+  #elif IS_RRD_FG_SC
+
+    #define BEEPER_PIN                       151
+
+    //#define LCD_PINS_D5                    -1
+    //#define LCD_PINS_D6                    -1
+    //#define LCD_PINS_D7                    -1
+
+  #else
+
+    //#error "Only CR10_STOCKDISPLAY and REPRAP_DISCOUNT_FULL_GRAPHIC_SMART_CONTROLLER are currently supported. Comment out this line to continue."
+
+  #endif
+
+  #define BTN_EN1                             2
+  #define BTN_EN2                             4
+  #define BTN_ENC                             12
+  #define BEEPER_PIN                          139
+#endif // HAS_MARLINUI_U8GLIB
+
+// Hardware serial pins
+// Add the following to Configuration.h or Configuration_adv.h to assign
+// specific pins to hardware Serial1 and Serial2.
+// Note: Serial2 can be defined using HARDWARE_SERIAL2_RX and HARDWARE_SERIAL2_TX but
+// MRR ESPA does not have enough spare pins for such reassignment.
+//#define HARDWARE_SERIAL1_RX                 21
+//#define HARDWARE_SERIAL1_TX                 22
+//#define HARDWARE_SERIAL2_RX                  2
+//#define HARDWARE_SERIAL2_TX                  4
+
+//
+// Filament Runout Sensor -1关闭同时注释Configuration.h 1288 
+//
+#ifndef FIL_RUNOUT_PIN
+  #define FIL_RUNOUT_PIN                      13
+#endif
+//
+// Servos 3Douch探针
+//
+#define SERVO0_PIN                            14
+//探针独立，不用于Z限位
+//Configuration.h里929行取消注释//#define Z_MIN_PROBE_PIN 35 // Pin 35 is the RAMPS default
+
+//
+// M3/M4/M5 - Spindle/Laser Control激光
+//
+#define SPINDLE_LASER_ENA_PIN              -1// FET 1
+#define SPINDLE_LASER_PWM_PIN              33// Bed FET
+//#define SPINDLE_DIR_PIN                    -1// FET 3
+
+//
+//拓展板
+//
+//+2只有2个MAX限位，+5的限位和热敏只能6选3，开启需注释掉46行Limit Switches
+#ifndef X_STOP_PIN
+  #ifndef X_MIN_PIN
+    #define X_MIN_PIN                          15
+  #endif
+  #ifndef X_MAX_PIN
+    #define X_MAX_PIN                          -1//32
```

Comment:
Is there any situation with this board where pins 32, 33, 34 are expected to be used as endstops?